common/containers/Tabs/SignAndVerifyMessage/components/SignMessage/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { AppState } from 'reducers';
+import './index.scss';
+
+export interface Props {
+  wallet: IWallet;
+  showNotification: TShowNotification;
+}
+
+export interface State {
+  message: string;
+  signMessageError: string;
+  signedMessage: ISignedMessage | null;
+}
+
+export const initialState: State = {
+  message: '',
+  signMessageError: '',
+  signedMessage: null
+};
+
+const messagePlaceholder =
+  'This is a sweet message that you are signing to prove that you own the address you say you own.';
+
+export class SignMessage extends Component<Props, State> {
+  public state: State = initialState;
+
+  public render() {
+    const { wallet } = this.props;
+
+    return (
+      <div>
+        <div className="Tab-content-pane">
+          <h4>{translate('MSG_message')}</h4>
+          <div className="form-group">
+            <textarea
+              className={`SignMessage-inputBox form-control ${!!this.state
+                .message
+                ? 'is-valid'
+                : 'is-invalid'}`}
+              placeholder={messagePlaceholder}
+              value={this.state.message}
+              onChange={this.handleMessageChange}
+            />
+            <div className="SignMessage-help">{translate('MSG_info2')}</div>
+          </div>
+
+          {!!this.props.wallet && (
+            <button
+              className="SignMessage-sign btn btn-primary btn-lg"
+              onClick={this.handleSignMessage}
+              disabled={false}
+            >
+              {translate('NAV_SignMsg')}
+            </button>
+          )}
+
+          {!!this.state.signedMessage && (
+            <div>
+              <h4>{translate('MSG_signature')}</h4>
+              <div className="form-group">
+                <textarea
+                  className={'SignMessage-inputBox form-control'}
+                  value={JSON.stringify(this.state.signedMessage, null, 2)}
+                  disabled={true}
+                  onChange={this.handleMessageChange}
+                />
+              </div>
+            </div>
+          )}
+        </div>
+
+        {!wallet && <WalletDecrypt />}
+      </div>
+    );
+  }
+
+  private handleSignMessage = async () => {
+    const { wallet } = this.props;
+    const { message } = this.state;
+
+    if (!wallet) {
+      return;
+    }
+
+    try {
+      const signedMessage: ISignedMessage = {
+        address: await wallet.getAddress(),
+        message,
+        signature: await wallet.signMessage(message),
+        version: '2'
+      };
+
+      this.setState({ signedMessage });
+      this.props.showNotification(
+        'success',
+        `Successfully signed message with address ${signedMessage.address}.`
+      );
+    } catch (err) {
+      console.error(err);
+      this.props.showNotification(
+        'danger',
+        `Error signing message: ${err.message}`
+      );
+    }
+  };
+
+  private handleMessageChange = (
+    e: React.SyntheticEvent<HTMLTextAreaElement>
```

Comment:
Im assuming  you got this typing from the typescript refactor PR, turns out we did the typing wrong there. It should be `e: React.FormEvent<HTMLTextAreaElement>`
Then message would be `const message = e.currentTarget.value`